### PR TITLE
Cs 5999/share current context with messages user gated

### PR DESCRIPTION
**Pull request by IanCal**
Builds on https://github.com/cardstack/boxel/pull/709

This adds a checkbox (see lower right) that means all messages sent come along with a "context" that shares largely what the user can see. This is *only* sent when a message is sent, but means the bot can have an up-to-date view of what the user is doing. 

Currently shares the submode (interact) and open cards (only the top of the stack). 

image

### CODE REVIEW

**Comment by tintinthong** [comment]
Hmmm how would I know if the bot is taking into account the context of the cards in the stack? What difference would the replies be?

**Comment by tintinthong** [comment]
> Hmmm how would I know if the bot is taking into account the context of the cards in the stack? What difference would the replies be? Something like this?

File: packages/host/app/components/matrix/room.gts [hunk 1]
```diff
@@ -173,12 +177,14 @@ export default class Room extends Component<RoomArgs> {
...
           {{on 'click' this.chooseCard}}
         >Choose Card</Button>
-        <Button
-          data-test-send-open-cards-btn
-          @loading={{this.doSendMessage.isRunning}}
-          @disabled={{this.doSendMessage.isRunning}}
-          {{on 'click' this.sendOpenCards}}
-        >Send open cards</Button>
+
+        <label for='share-checkbox'>Share top cards</label>
```
Comment (tintinthong) [inline packages/host/app/components/matrix/room.gts]: do you mean? "Share cards in stack"?

Comment (tintinthong) [reply packages/host/app/components/matrix/room.gts]: Then again, I am not sure if the user will know what it even does? Im not too sure what results I am expecting from this PR from the UI

Comment (IanCal) [reply packages/host/app/components/matrix/room.gts]: Yeah we don't have design yet for this, I'm happy to make any changes to make it clearer before then.

Comment (IanCal) [reply packages/host/app/components/matrix/room.gts]: > do you mean? "Share cards in stack"?

This shares only the cards at the top of the stack, I'll change the wording there as it's not clear

Comment (tintinthong) [reply packages/host/app/components/matrix/room.gts]: Ok sounds good.

**Comment by jurgenwerk** [comment]
I took this for a spin and it looks like the bot indeed has the context for the currently opened card. I was able to get it to write and update the blogposts's body: 

image

I didn't test other stuff as I'm a little unclear on what is the expected goal here. The sidebar looks quite messy so it's confusing to test for me (especially since this was my first time running the bot 😀). Also another thing I am wondering about is how come the sharing context isn't simply enabled by default? Isn't the goal of the bot to help to come up with content for the cards?

**Comment by IanCal** [comment]
@jurgenwerk thanks for taking it for a spin.

> The sidebar looks quite messy so it's confusing to test for me (especially since this was my first time running the bot 😀)

Yeah there's design in progress, what's making this harder is that currently it's a general chat *and* this AI tool mixed together. As we don't yet have a clear design I've tried to minimise changes. Consequences of how this started I think.

Could be a task for the next sprint, either removing the other chat parts, or at least splitting out the AI stuff into a distinct component that doesn't have things like room objectives, invites, etc. That could be done before there's design work.

>  Also another thing I am wondering about is how come the sharing context isn't simply enabled by default? Isn't the goal of the bot to help to come up with content for the cards?

It's a good question - the initial idea here is that we didn't want to take your private data and send it off without active permission. A broader goal of this interface is that it'll be able to do more than this task and how much of your current context you're happy sharing 

A first step on that direction would be adding coding help when you open this in the code mode. Answering general typescript or Card development questions without seeing your code would be useful alone (give an example computed) - with access to the code it could help make a more relevant change.

**Comment by IanCal** [comment]
There were some merge conflicts, changes to main and some conflicts with https://github.com/cardstack/boxel/pull/726 - I've pulled those commits into this branch too, so that if they're both ok to go we can merge more smoothly.

File: packages/host/app/components/matrix/room.gts (hunk 1 above)
Comment (IanCal) [reply packages/host/app/components/matrix/room.gts]: I've made this probably overly specific, but hopefully clearer for development & intent until there's a full design.

**Review by tintinthong — APPROVED** [review]
👍

**Pull request merged**